packages/hdwallet-xdefi/src/xdefi.ts

```diff
@@ -63,6 +65,21 @@ export class XDEFIHDWalletInfo implements core.HDWalletInfo, core.ETHWalletInfo
     return chainId === 1;
   }
 
+  public async ethSwitchChain(chainId = 1): Promise<void> {
+    const hexChainId = ethers.utils.hexValue(chainId);
+    const provider: any = await detectEthereumProvider({ mustBeMetaMask: true, silent: false, timeout: 3000 });
+    try {
+      // at this point, we know that we're in the context of a valid MetaMask provider
+      await provider.request({ method: "wallet_switchEthereumChain", params: [{ chainId: hexChainId }] });
+    } catch (e: any) {
+      const error: core.SerializedEthereumRpcError = e;
+      if (error.code === 4902) {
+        // TODO: EVM Chains Milestone
+        // We will need to pass chainName and rpcUrls, which we don't have yet, to add a chain to XDEFI.
+      }
+    }
+  }
+
```

Comment:
I'm not sure we need this method added to the XDEFIHDWalletInfo class.

Reply:
Seems like we don't need it there, the call worked without. Removed in de254daa059182713ac2e6c2d17869ff1fc55840